### PR TITLE
Avoid warning: Extract first element explicitly

### PR DESCRIPTION
**Pull request by maurolepore**
Relates to #436 

When fed with an object of length > 1L, `if()` extracts the first element with a warning. This PR removes such warning in the most backward-compatible way -- by extracting the fist element explicitly. but it also flags `FIXME` because this is likely incorrect and instead we should likely use get an atomic vector with `all()`.

```r
# Okay (backward compatible but likely wrong)
if (long_logical[[1]]) ...

# Likely better
if (all(long_logical)) ...

# Bad
if (long_logical) ...
```

### CODE REVIEW

**Review by cjyetman — COMMENTED** [review]
Thanks. I'd prefer to dig a bit deeper and try to understand what the intent was and do it properly rather than add a band-aid though. Can we do that?

File: 0_web_functions.R [hunk 1]
```diff
@@ -318,7 +318,13 @@ save_cleaned_files <- function(save_loc,
...
   fst::write_fst(company_emissions, file.path(save_loc, "company_emissions.fst"))
-  if (!is.na(total_fund_list))
+  # FIXME: Looking at the signature and noticing this is the only use of
+  # `total_fund_list` I interpret that the author intended to make
+  # `total_fund_list` an optional argument and chose as a default the value
+  # `NA`. Here I express that intent explicitely but note that not `NA` but
+  # `NULL` is more commonly used as a default value for optional arguments.
+  total_fund_list_not_passed_as_argument <- identical(total_fund_list, NA)
+  if (!is.na(total_fund_list_not_passed_as_argument))
   {fst::write_fst(total_fund_list, file.path(save_loc, "total_fund_list.fst"))}
```
Comment (maurolepore) [inline 0_web_functions.R:328]: Same here, although I don't know if / where this code is used.

**Comment by cjyetman** [comment]
I agree with your assessment in the comment... this was intended to be a non-mandatory argument and the default should be `NULL`... so my preference would be to fix the function signature to do it properly and adapt the calls to it appropriately... I think that's better than jumping through hoops to get `NA` to work without warnings and leaving comments in the code explaining why that will probably stay there forever.

**Comment by maurolepore** [comment]
Great. I did that in two commits. The first one just implementing the changes; the second styling the code for readability. I can split that in a separate PR if you prefer, but I think the style change is small and doubt will make this PR hard to understand.

File: 0_portfolio_input_check_functions.R [hunk 2]
```diff
@@ -1085,13 +1085,7 @@ process_raw_portfolio <- function(portfolio_raw,
...
 
   # correct Funds classification by comparing isin to the list of all known funds isins
-  # FIXME: Looking at the signature and noticing this is the only use of
-  # `total_fund_list` I interpret that the author intended to make
-  # `total_fund_list` an optional argument and chose as a default the value
-  # `NA`. Here I express that intent explicitely but note that not `NA` but
-  # `NULL` is more commonly used as a default value for optional arguments.
-  total_fund_list_not_passed_as_argument <- identical(total_fund_list, NA)
-  if(total_fund_list_not_passed_as_argument){portfolio <- portfolio %>% mutate(asset_type = ifelse(is.element(isin, total_fund_list$fund_isin), "Funds", asset_type))}
+  if(!is.null(total_fund_list)){portfolio <- portfolio %>% mutate(asset_type = ifelse(is.element(isin, total_fund_list$fund_isin), "Funds", asset_type))}
```
Comment (maurolepore) [inline 0_portfolio_input_check_functions.R]: Please review my use us `!is.null()` -- I tend to get confused with this kind of double negatives.

Comment (cjyetman) [reply 0_portfolio_input_check_functions.R]: it looks right to me.... but it looks like the original change with `total_fund_list_not_passed_as_argument` was wrong (i.e. it should **not** do the thing if `total_fund_list` was **not** passed). Now I'm confused... I'll have to look closer at it tomorrow or Wednesday.

Comment (maurolepore) [reply 0_portfolio_input_check_functions.R]: I think you are right in that my previous code was wrong. So confusing :-)

**Review by cjyetman — APPROVED** [review]

**Pull request merged**